### PR TITLE
Filter FP strings

### PR DESCRIPTION
**Pull request by mr-tz**
Creating this PR to open the discussion.
The goal of these filters is to reduce the FP rate. However, it is possible that we filter out true positive strings. Should we add a command-line argument to let users disable filtering (enable it by default)? We then probably should move the filtering to the print_* functions.

### CODE REVIEW

**Comment by williballenthin** [comment]
changes look good. i like how were are separating out the specific details into readable functions and namespacing them.

**Comment by williballenthin** [comment]
> Should we add a command-line argument to let users disable filtering (enable it by default)

exactly

> We then probably should move the filtering to the print_* functions.

ok, seems like a fair place to put this

File: floss/utils.py [hunk 1]
```diff
@@ -76,3 +77,26 @@ def get_vivisect_meta_info(vw, selected_functions):
...
+    return FP_FILTER_CHARS.match(s)
+
+
+def strip_string(s):
+    """
+    Return string stripped from false positive (FP) pre- or suffixes.
+    :param s: input string
+    :return: string stripped from FP pre- or suffixes
+    """
+    return re.sub(FP_FILTER_PREFIXES, "", re.sub(FP_FILTER_SUFFIXES, "", s))
```
Comment (mr-tz) [inline floss/utils.py]: See 8df5b2f

**Comment by mr-tz** [comment]
If we move filtering to the print functions other tools that use FLOSS won't benefit from this change. I lean towards keeping the filtering in the `extract_*` methods. Alternatively, we could add a separate `filter_strings` function that takes all decoded/stack strings and returns the filtered and stripped strings.

**Comment by mr-tz** [comment]
I'm not sure if it's my machine, but FLOSS runs very slow (>7 minutes) for me at the moment...

**Comment by mr-tz** [comment]
@williballenthin should we bump up the version (the API changed a little bit) and merge?

**Comment by williballenthin** [comment]
yup, lets do it

**Pull request merged**